apicast/src/proxy.lua

```diff
@@ -361,6 +362,15 @@ function _M:access(service)
 
   local credentials, err = service:extract_credentials()
 
+  if keycloak then
```

Comment:
Yep. Here it is well isolated. It can wait here for a while until we figure out better place.

But still it should be executed only when the auth method is oauth.